pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/rest/BridgeResource.java

```diff
@@ -158,6 +158,11 @@ public void write(final OutputStream out) throws IOException,
                     if (!threadSafe) {
                         unlock(dataDir);
                     }
+                    try {
+                        bridge.endIteration();
```

Comment:
should we unlock after closing resources, at the very end ? lock --> start --> end --> unlock ?